src/functions-and-arrays.js

```diff
@@ -1,41 +1,95 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1
+  }
+  else {
+    return number2
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length === 0) { return null }
+  let longestWord = words.sort(function(a, b) { return b.length - a.length; });
+  return longestWord[0];
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(isArray) {
+  let sum = 0;
+
+for (let i = 0; i < isArray.length; i++) {
+    sum += isArray[i];
+}
+return(sum);
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(arr) {
+  if (arr.length === 0) { return 0 }
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(isArray) {
+  if (isArray.length === 0) { return null }
+
+let sum = 0;
+let total;
+
+  for (let i = 0; i < isArray.length; i++) {
+    sum += isArray[i];
+    total = sum / isArray.length;
+}
+return(total);
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(isArray) { 
+  let arrayOfLength = [];
+  if (isArray.length === 0) { return null }
+
+  
+  for (let i = 0; i < isArray.length; i++) {
+  
+   arrayOfLength.push(isArray[i].length);
+   }
+
+  let sum = 0;
+  let total;
+
+  for (let i = 0; i < arrayOfLength.length; i++) {
+    sum += arrayOfLength[i];
+    total = sum / arrayOfLength.length;
```

Comment:
`total = sum / arrayOfLength `

should go outside the for loop, because you should do the division when you have the total amount in sum